src/core/server/saved_objects/migrationsv2/migrations_state_action_machine.ts

```diff
@@ -180,6 +193,7 @@ export async function migrationStateActionMachine({
       throw new Error('Invalid terminating control state');
     }
   } catch (e) {
+    await cleanup(client, executionLog, lastState);
```

Comment:
not really happy to put cleanup logic here, but it seems to be the only option considering `migrationStateActionMachine` throws exceptions as well

Reply:
> I wonder if we could change the behavior of stateActionMachine to transition to FATAL

Agree, I also expect the exception to transition FSA into `FATAL` state. But it's definitely out of the scope of the current PR.  Not sure we need to refactor it right now, but we can create an issue to pay the tech debt off later.